### PR TITLE
[libc] Add `vsscanf` function

### PR DESCRIPTION
**Pull request by izaakschroeder**
Still needs proper implementation, but stubs are here.

See:
- https://linux.die.net/man/3/vsscanf
- https://android.googlesource.com/platform/bionic/+/froyo-release/libc/stdio/vsscanf.c

### CODE REVIEW

**Review by jhuber6 — COMMENTED** [review]
The implementation is trivial here, can you go ahead and do it? All you need to do is copy the code from `sscanf` sans the `va_list` creation.

**Comment by izaakschroeder** [comment]
@jhuber6 did my best! 😅

**Review by jhuber6 — COMMENTED** [review]
Thanks, now we just need a test. I think you should add this to every `config/*/entrypoints.txt` that already contains `sscanf`. The test should then add `vsscanf` as a dependency in the CMakeLists.txt. Add a single test that calls `vsscanf` using an existing va_list.

File: libc/spec/stdc.td [hunk 1]
```diff
@@ -968,6 +968,13 @@ def StdC : StandardSpec<"stdc"> {
               RetValSpec<IntType>,
               [ArgSpec<IntType>, ArgSpec<FILEPtr>]
           >,
+          FunctionSpec<
```
Comment (michaelrj-google) [inline libc/spec/stdc.td:971]: you should also add `vsscanf` to `entrypoints.txt` for the relevant platforms

**Review by jhuber6 — APPROVED** [review]
Looks fine as is, but I'd change the invocations like
```c
auto fn = [](const char *str, const char *fmt, ...) -> int { ... };
fn("Not an Integer", %d, &result);
```

**Comment by jhuber6** [comment]
Can you make the above suggestions and add this to the entrypoints.txt for every target that already contains `sscanf`?

**Comment by jhuber6** [comment]
merged in https://github.com/llvm/llvm-project/pull/101402, thanks for the initial patch.

**Pull request closed without merging**